ReactiveCocoaTests/Swift/SignalSpec.swift

```diff
@@ -14,50 +14,293 @@ import ReactiveCocoa
+				// expect.
+				let test: () -> () = {
+					let (signal, observer) = Signal<(), TestError>.pipe()
+					weakSignal = signal
+					testScheduler.schedule {
+						sendCompleted(observer)
+					}
+				}
+				test()
+				
+				expect(weakSignal).toNot(beNil())
+				
+				testScheduler.run()
+				expect(weakSignal).to(beNil())
 			}
 
-			pending("should forward events to observers") {
+			it("should forward events to observers") {
+				let (signal, observer) = Signal<Int, NoError>.pipe()
+				
+				var fromSignal: [Int] = []
+				var completed = false
+				
+				signal.observe(next: { number in
+					fromSignal.append(number)
+				}, completed: {
+					completed = true
+				})
+				
+				expect(fromSignal).to(beEmpty())
+				expect(completed).to(beFalsy())
+				
+				sendNext(observer, 1)
+				expect(fromSignal).to(equal([1]))
+				
+				sendNext(observer, 2)
+				expect(fromSignal).to(equal([1, 2]))
+				
+				expect(completed).to(beFalsy())
+				sendCompleted(observer)
+				expect(completed).to(beTruthy())
 			}
 		}
 
 		describe("observe") {
-			pending("should stop forwarding events when disposed") {
+			var testScheduler: TestScheduler!
+			
+			beforeEach {
+				testScheduler = TestScheduler()
+			}
+			
+			it("should stop forwarding events when disposed") {
+				let disposable = SimpleDisposable()
+				
+				let signal: Signal<Int, NoError> = Signal { observer in
+					testScheduler.schedule {
+						for number in [1, 2] {
+							sendNext(observer, number)
+						}
+						sendCompleted(observer)
+						sendNext(observer, 4)
+					}
+					return disposable
+				}
+				
+				var fromSignal: [Int] = []
+				signal.observe(next: { number in
+					fromSignal.append(number)
+				})
+				
+				expect(disposable.disposed).to(beFalsy())
+				expect(fromSignal).to(beEmpty())
+				
+				testScheduler.run()
+				
+				expect(disposable.disposed).to(beTruthy())
+				expect(fromSignal).to(equal([1, 2]))
 			}
 
-			pending("should not trigger side effects") {
+			it("should not trigger side effects") {
+				var runCount = 0
+				let signal: Signal<(), NoError> = Signal { observer in
+					runCount += 1
+					return nil
+				}
+				
+				expect(runCount).to(equal(1))
+				
+				signal.observe()
+				expect(runCount).to(equal(1))
 			}
 
-			pending("should release observer after termination") {
+			it("should release observer after termination") {
+				weak var testStr: NSMutableString?
+				let (signal, sink) = Signal<Int, NoError>.pipe()
+
+				let test: () -> () = {
+					var innerStr: NSMutableString = NSMutableString()
+					signal.observe(next: { value in
+						innerStr.appendString("\(value)")
+					})
+					testStr = innerStr
+				}
+				test()
+
+				sendNext(sink, 1)
+				expect(testStr).to(equal("1"))
+				sendNext(sink, 2)
+				expect(testStr).to(equal("12"))
+
+				sendCompleted(sink)
+				expect(testStr).to(beNil())
 			}
 
 			pending("should release observer after disposal") {
```

Comment:
@jspahrsummers what is difference between this and previous (`"should release observer after termination"`)  test?